core/domain/exp_domain.py

```diff
@@ -2977,31 +2977,31 @@ def get_state_names_mapping(self, change_list):
                 commit.
 
         Returns:
-            dict. The state_names_mapping dict.
+            dict. The new_to_old_state_names mapping dict.
         """
-        state_names_mapping = {}
+        old_to_new_state_names = {}
 
         for state_name in self.states:
-            state = self.states[state_name]
-            state_names_mapping[state_name] = state_name
+            old_to_new_state_names[state_name] = state_name
 
         for change_dict in reversed(change_list):
             if change_dict['cmd'] == CMD_RENAME_STATE:
-                state_names_mapping[change_dict['old_state_name']] = (
-                    state_names_mapping.pop(change_dict['new_state_name']))
+                old_to_new_state_names[change_dict['old_state_name']] = (
+                    old_to_new_state_names.pop(change_dict['new_state_name']))
 
-        state_names_mapping = {value: key for key, value in (
-            state_names_mapping.items())}
-        return state_names_mapping
+        new_to_old_state_names = {
+            value: key for key, value in old_to_new_state_names.items()
```

Comment:
Btw it's better here to use .iteritems() rather than items() (the latter creates a separate construct and the former just iterates).